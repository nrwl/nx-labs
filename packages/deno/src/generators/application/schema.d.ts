export interface ApplicationGeneratorSchema {
    name: string;
    tags?: string;
    directory?: string;
}