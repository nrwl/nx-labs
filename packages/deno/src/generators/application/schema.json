{
  "$schema": "http://json-schema.org/schema",
  "$id": "Application",
  "title": "Generate an Deno application.",
  "description": "Setup a new Deno application in the workspace.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the application.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?",
      "pattern": "^[a-zA-Z].*$"
    },
    "unitTestRunner": {
      "type": "string",
      "description": "Add a testing target for this application.",
      "enum": ["deno", "none"],
      "default": "deno"
    },
    "linter": {
      "type": "string",
      "description": "Add a linting target for this application.",
      "enum": ["deno", "none"],
      "default": "deno"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d",
      "x-priority": "important"
    },
    "withWatch": {
      "type": "boolean",
      "description": "Default the project to watch for file changes for the 'serve' target. For projects that do not need watch capability (eg: CLI projects), use --with-watch=false",
      "default": true
    },
    "monorepo": {
      "type": "boolean",
      "description": "Creates an integrated monorepo.",
      "aliases": ["integrated"],
      "x-priority": "internal"
    },
    "rootProject": {
      "type": "boolean",
      "x-priority": "internal"
    },
    "framework": {
      "type": "string",
      "description": "Generate an app with a framework already setup",
      "enum": ["oak", "none"]
    }
  },
  "required": ["name"]
}
