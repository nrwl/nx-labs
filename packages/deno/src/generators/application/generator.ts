import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  Tree,
} from '@nrwl/devkit';
import { join } from 'path';
import { initDeno } from '../init/generator';
import { addPathToDenoSettings } from '../utils/add-path';
import { ApplicationGeneratorSchema } from './schema';

interface NormalizedSchema extends ApplicationGeneratorSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

function normalizeOptions(
  tree: Tree,
  options: ApplicationGeneratorSchema
): NormalizedSchema {
  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const projectRoot = `${getWorkspaceLayout(tree).appsDir}/${projectDirectory}`;
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  return {
    ...options,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    importMapPath: joinPathFragments(
      offsetFromRoot(options.projectRoot),
      'import_map.json'
    ),
    template: '',
  };
  generateFiles(
    tree,
    join(__dirname, 'files'),
    options.projectRoot,
    templateOptions
  );
}

export default async function (
  tree: Tree,
  options: ApplicationGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, options);
  initDeno(tree);
  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.projectRoot,
    projectType: 'application',
    name: normalizedOptions.projectName,
    sourceRoot: `${normalizedOptions.projectRoot}/src`,
    targets: {
      build: {
        executor: '@nrwl/deno:bundle',
        outputs: [`dist/${normalizedOptions.projectRoot}`],
        options: {
          main: `${normalizedOptions.projectRoot}/src/main.ts`,
          outputFile: `dist/${normalizedOptions.projectRoot}/main.js`,
          denoConfig: `${normalizedOptions.projectRoot}/deno.json`,
        },
      },
      serve: {
        executor: '@nrwl/deno:run',
        options: {
          buildTarget: `${normalizedOptions.projectName}:build`,
        },
      },
      test: {
        executor: '@nrwl/deno:test',
        outputs: [`coverage/${normalizedOptions.projectRoot}`],
        options: {
          coverageDirectory: `coverage/${normalizedOptions.projectRoot}`,
          denoConfig: `${normalizedOptions.projectRoot}/deno.json`,
        },
      },
      lint: {
        executor: '@nrwl/deno:lint',
        options: {
          denoConfig: `${normalizedOptions.projectRoot}/deno.json`,
        },
      },
    },
    tags: normalizedOptions.parsedTags,
  });
  addFiles(tree, normalizedOptions);
  addPathToDenoSettings(tree, normalizedOptions.projectRoot);
  await formatFiles(tree);
}
