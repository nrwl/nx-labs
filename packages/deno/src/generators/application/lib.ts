import {
  addProjectConfiguration,
  generateFiles,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  ProjectConfiguration,
  Tree,
} from '@nx/devkit';
import { join } from 'path';
import { DenoAppGeneratorSchema, DenoAppNormalizedSchema } from './schema';

export function normalizeOptions(
  tree: Tree,
  options: DenoAppGeneratorSchema
): DenoAppNormalizedSchema {
  // --monorepo takes precedence over --rootProject
  // This is for running `create-nx-workspace --preset=@nx/deno --monorepo`
  const rootProject = !options.monorepo && options.rootProject;

  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const appDir = getWorkspaceLayout(tree).appsDir;
  // prevent paths from being dist/./app-name
  const projectRoot = rootProject
    ? '.'
    : joinPathFragments(appDir === '.' ? '' : appDir, projectDirectory);
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  options.framework ??= 'none';

  return {
    ...options,
    rootProject,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
  };
}

export function addFiles(tree: Tree, options: DenoAppNormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    importMapPath: joinPathFragments(
      offsetFromRoot(options.projectRoot),
      'import_map.json'
    ),
    template: '',
  };
  generateFiles(
    tree,
    join(__dirname, 'files', 'root-files'),
    options.projectRoot,
    templateOptions
  );

  generateFiles(
    tree,
    join(__dirname, 'files', `framework-${options.framework}`),
    join(options.projectRoot, 'src'),
    templateOptions
  );
}

export function addProjectConfig(tree: Tree, opts: DenoAppNormalizedSchema) {
  const coverageDirectory = joinPathFragments(
    'coverage',
    opts.rootProject ? opts.name : opts.projectRoot
  );
  const targets: ProjectConfiguration['targets'] = {
    build: {
      executor: '@nx/deno:emit',
      outputs: [
        joinPathFragments(
          'dist',
          opts.rootProject ? opts.name : opts.projectRoot
        ),
      ],
      options: {
        main: joinPathFragments(opts.projectRoot, 'src/main.ts'),
        outputFile: joinPathFragments(
          'dist',
          opts.rootProject ? opts.name : opts.projectRoot,
          'main.js'
        ),
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
    serve: {
      executor: '@nx/deno:run',
      options: {
        buildTarget: `${opts.projectName}:build`,
      },
    },
    test: {
      executor: '@nx/deno:test',
      outputs: [coverageDirectory],
      options: {
        coverageDirectory,
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
    lint: {
      executor: '@nx/deno:lint',
      options: {
        denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
      },
    },
  };

  if (opts.withWatch === true) {
    targets.serve.options.watch = true;
  }

  if (opts.linter === 'none') {
    delete targets.lint;
  }

  if (opts.unitTestRunner === 'none') {
    delete targets.test;
  }

  addProjectConfiguration(tree, opts.projectName, {
    root: opts.projectRoot,
    projectType: 'application',
    name: opts.projectName,
    sourceRoot: joinPathFragments(opts.projectRoot, 'src'),
    targets,
    tags: opts.parsedTags,
  });
}
