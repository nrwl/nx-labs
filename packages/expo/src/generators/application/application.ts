import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import {
  convertNxGenerator,
  Tree,
  formatFiles,
  joinPathFragments,
  GeneratorCallback,
} from '@nrwl/devkit';

import { addLinting } from '../../utils/add-linting';
import { addJest } from '../../utils/add-jest';
import { runSymlink } from '../../utils/symlink-task';

import { normalizeOptions } from './lib/normalize-options';
import initGenerator from '../init/init';
import { addProject } from './lib/add-project';
import { addDetox } from './lib/add-detox';
import { createApplicationFiles } from './lib/create-application-files';
import { Schema } from './schema';

export async function expoApplicationGenerator(
  host: Tree,
  schema: Schema
): Promise<GeneratorCallback> {
  const options = normalizeOptions(schema);

  createApplicationFiles(host, options);
  addProject(host, options);

  const initTask = await initGenerator(host, { ...options, skipFormat: true });
  const lintTask = await addLinting(
    host,
    options.projectName,
    options.appProjectRoot,
    [joinPathFragments(options.appProjectRoot, 'tsconfig.app.json')],
    options.linter,
    options.setParserOptionsProject
  );
  const jestTask = await addJest(
    host,
    options.unitTestRunner,
    options.projectName,
    options.appProjectRoot,
    options.js
  );
  const detoxTask = await addDetox(host, options);
  const symlinkTask = runSymlink(host.root, options.appProjectRoot);

  if (!options.skipFormat) {
    await formatFiles(host);
  }

  return runTasksInSerial(initTask, lintTask, jestTask, detoxTask, symlinkTask);
}

export default expoApplicationGenerator;
export const expoApplicationSchematic = convertNxGenerator(
  expoApplicationGenerator
);
