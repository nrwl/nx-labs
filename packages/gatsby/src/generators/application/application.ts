import { convertNxGenerator, formatFiles, Tree } from '@nx/devkit';
import { runTasksInSerial } from '@nx/workspace/src/utilities/run-tasks-in-serial';

import { addStyleDependencies } from '../../utils/styles';
import { gatsbyInitGenerator } from '../init/init';
import { addCypress } from './lib/add-cypress';
import { addGitIgnoreEntry } from './lib/add-git-ignore-entry';
import { addJest } from './lib/add-jest';
import { addLinting } from './lib/add-linting';
import { addPrettierIgnoreEntry } from './lib/add-prettier-ignore-entry';
import { addProject } from './lib/add-project';
import { createApplicationFiles } from './lib/create-application-files';
import { normalizeOptions } from './lib/normalize-options';
import { setDefaults } from './lib/set-defaults';
import { updateJestConfig } from './lib/update-jest-config';
import { Schema } from './schema';

export async function applicationGenerator(host: Tree, schema: Schema) {
  const options = normalizeOptions(host, schema);

  const initTask = await gatsbyInitGenerator(host, {
    ...options,
    skipFormat: true,
  });

  createApplicationFiles(host, options);
  addProject(host, options);
  const styledTask = addStyleDependencies(host, options.style);
  const lintTask = await addLinting(host, options);
  const cypressTask = await addCypress(host, options);
  const jestTask = await addJest(host, options);
  updateJestConfig(host, options);
  addPrettierIgnoreEntry(host, options);
  addGitIgnoreEntry(host, options);

  setDefaults(host, options);
  await formatFiles(host);

  return runTasksInSerial(
    initTask,
    styledTask,
    lintTask,
    cypressTask,
    jestTask
  );
}

export default applicationGenerator;
export const applicationSchematic = convertNxGenerator(applicationGenerator);
