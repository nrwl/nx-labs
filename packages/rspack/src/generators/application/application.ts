import {
  addDependenciesToPackageJson,
  addProjectConfiguration,
  ensurePackage,
  formatFiles,
  generateFiles,
  joinPathFragments,
  names,
  offsetFromRoot as _offsetFromRoot,
  Tree,
} from '@nrwl/devkit';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { version as nxVersion } from 'nx/package.json';
import * as path from 'path';
import {
  lessVersion,
  reactDomVersion,
  reactVersion,
  sassVersion,
  stylusVersion,
  typesReactDomVersion,
  typesReactVersion,
} from '../../utils/versions';
import { configurationGenerator } from '../configuration/configuration';
import { addCypress } from './lib/add-cypress';
import { addJest } from './lib/add-jest';
import { addLinting } from './lib/add-linting';
import { createTsConfig } from './lib/create-ts-config';
import { normalizeOptions } from './lib/normalize-options';
import { ApplicationGeneratorSchema, NormalizedSchema } from './schema';

export default async function (
  tree: Tree,
  _options: ApplicationGeneratorSchema
) {
  await ensurePackage(tree, '@nrwl/react', nxVersion);
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const { reactInitGenerator } = require('@nrwl/react');

  const tasks = [];
  const options = normalizeOptions(tree, _options);

  options.style ??= 'css';

  addProjectConfiguration(tree, options.name, {
    root: options.appProjectRoot,
    projectType: 'application',
    sourceRoot: `${options.appProjectRoot}/src`,
    targets: {},
  });

  const offsetFromRoot = _offsetFromRoot(options.appProjectRoot);
  generateFiles(tree, path.join(__dirname, 'files'), options.appProjectRoot, {
    ...options,
    ...names(options.name),
    offsetFromRoot,
    template: '',
  });

  createTsConfig(
    tree,
    options,
    joinPathFragments(offsetFromRoot, 'tsconfig.base.json')
  );

  const projectTask = await configurationGenerator(tree, {
    project: options.name,
    devServer: true,
    tsConfig: joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
    uiFramework: 'react',
    target: 'web',
    main: joinPathFragments(options.appProjectRoot, 'src/main.tsx'),
  });
  tasks.push(projectTask);

  const jestTask = await addJest(tree, options);
  tasks.push(jestTask);

  const cypressTask = await addCypress(tree, options);
  tasks.push(cypressTask);

  const lintTask = await addLinting(tree, options);
  tasks.push(lintTask);

  const installTask = addDependenciesToPackageJson(
    tree,
    { react: reactVersion, 'react-dom': reactDomVersion },
    {
      ...getStyleDependency(options),
      '@nrwl/react': nxVersion,
      '@types/react': typesReactVersion,
      '@types/react-dom': typesReactDomVersion,
    }
  );
  tasks.push(installTask);

  const reactInitTask = await reactInitGenerator(tree, options);
  tasks.push(reactInitTask);

  await formatFiles(tree);

  return runTasksInSerial(...tasks);
}

function getStyleDependency(options: NormalizedSchema): Record<string, string> {
  if (options.style === 'scss') return { sass: sassVersion };
  if (options.style === 'less') return { less: lessVersion };
  if (options.style === 'styl') return { stylus: stylusVersion };
  return {};
}
