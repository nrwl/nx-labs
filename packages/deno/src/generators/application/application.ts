import { formatFiles, installPackagesTask, Tree } from '@nx/devkit';
import { initDeno } from '../init/generator';
import { addPathToDenoSettings } from '../utils/add-path';
import { addFiles, addProjectConfig, normalizeOptions } from './lib';
import { DenoAppGeneratorSchema } from './schema';

export async function denoApplicationGenerator(
  tree: Tree,
  options: DenoAppGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, options);

  await initDeno(tree);
  addProjectConfig(tree, normalizedOptions);
  addFiles(tree, normalizedOptions);
  addPathToDenoSettings(tree, normalizedOptions.projectRoot);

  await formatFiles(tree);
  return () => {
    installPackagesTask(tree);
  };
}

export default denoApplicationGenerator;
