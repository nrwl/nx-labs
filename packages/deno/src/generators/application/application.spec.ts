import { readJson, readProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { denoApplicationGenerator } from './application';

describe('Deno App Generator', () => {
  let tree: Tree;
  describe('standalone', () => {
    beforeEach(() => {
      tree = createTreeWithEmptyWorkspace();
    });

    it('should make a deno app', async () => {
      await denoApplicationGenerator(tree, {
        name: 'my-app',
        rootProject: true,
      });
      expect(readJson(tree, 'deno.json')).toEqual({
        importMap: 'import_map.json',
      });
      expect(readProjectConfiguration(tree, 'my-app')).toMatchSnapshot();
      expect(tree.exists('src/main.ts')).toBeTruthy();
    });

    it('should make an oak api with --framework=oak', async () => {
      await denoApplicationGenerator(tree, {
        name: 'my-oak-api',
        rootProject: true,
        framework: 'oak',
      });

      expect(readJson(tree, 'deno.json')).toEqual({
        importMap: 'import_map.json',
      });
      expect(readProjectConfiguration(tree, 'my-oak-api')).toMatchSnapshot();
      expect(tree.read('src/main.ts', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('integrated', () => {
    beforeEach(() => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    });

    it('should make an integrated deno app', async () => {
      await denoApplicationGenerator(tree, {
        name: 'my-app',
      });
      expect(readJson(tree, 'apps/my-app/deno.json')).toEqual({
        importMap: '../../import_map.json',
      });
      expect(readProjectConfiguration(tree, 'my-app')).toMatchSnapshot();
      expect(tree.exists('apps/my-app/src/main.ts')).toBeTruthy();
    });

    it('should make an oak api with --framework=oak', async () => {
      await denoApplicationGenerator(tree, {
        name: 'my-oak-api',
        framework: 'oak',
      });
      expect(readJson(tree, 'apps/my-oak-api/deno.json')).toEqual({
        importMap: '../../import_map.json',
      });
      expect(readProjectConfiguration(tree, 'my-oak-api')).toMatchSnapshot();
      expect(
        tree.read('apps/my-oak-api/src/main.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });
});
