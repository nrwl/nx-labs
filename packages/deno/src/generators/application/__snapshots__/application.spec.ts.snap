// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deno App Generator integrated should make an integrated deno app 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "my-app",
  "projectType": "application",
  "root": "apps/my-app",
  "sourceRoot": "apps/my-app/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "apps/my-app/deno.json",
        "main": "apps/my-app/src/main.ts",
        "outputFile": "dist/apps/my-app/main.js",
      },
      "outputs": Array [
        "dist/apps/my-app",
      ],
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "apps/my-app/deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "my-app:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/apps/my-app",
        "denoConfig": "apps/my-app/deno.json",
      },
      "outputs": Array [
        "coverage/apps/my-app",
      ],
    },
  },
}
`;

exports[`Deno App Generator integrated should make an oak api with --framework=oak 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "my-oak-api",
  "projectType": "application",
  "root": "apps/my-oak-api",
  "sourceRoot": "apps/my-oak-api/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "apps/my-oak-api/deno.json",
        "main": "apps/my-oak-api/src/main.ts",
        "outputFile": "dist/apps/my-oak-api/main.js",
      },
      "outputs": Array [
        "dist/apps/my-oak-api",
      ],
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "apps/my-oak-api/deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "my-oak-api:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/apps/my-oak-api",
        "denoConfig": "apps/my-oak-api/deno.json",
      },
      "outputs": Array [
        "coverage/apps/my-oak-api",
      ],
    },
  },
}
`;

exports[`Deno App Generator integrated should make an oak api with --framework=oak 2`] = `
"import { Application, Router } from 'https://deno.land/x/oak@v11.1.0/mod.ts';

const router = new Router();

router.get('/', (ctx) => {
  ctx.response.redirect('/api');
});

router.get('/api', (ctx) => {
  ctx.response.body = { message: 'Hello my-oak-api' };
  ctx.response.type = 'text/json';
});

const app = new Application();

app.addEventListener('listen', ({ hostname, port, secure }) => {
  console.log(
    \`Listening on: \${secure ? 'https://' : 'http://'}\${
      hostname ?? 'localhost'
    }:\${port}\`
  );
});

app.use(router.routes());
app.use(router.allowedMethods());

await app
  .listen({ port: Number(Deno.env.get('PORT') || 4200) })
  .catch((err) => {
    console.error('Error serving app. Original Error:', err);
  });
"
`;

exports[`Deno App Generator standalone should make a deno app 1`] = `
Object {
  "$schema": "node_modules/nx/schemas/project-schema.json",
  "name": "my-app",
  "projectType": "application",
  "root": ".",
  "sourceRoot": "src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "deno.json",
        "main": "src/main.ts",
        "outputFile": "dist/my-app/main.js",
      },
      "outputs": Array [
        "dist/my-app",
      ],
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "my-app:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/my-app",
        "denoConfig": "deno.json",
      },
      "outputs": Array [
        "coverage/my-app",
      ],
    },
  },
}
`;

exports[`Deno App Generator standalone should make an oak api with --framework=oak 1`] = `
Object {
  "$schema": "node_modules/nx/schemas/project-schema.json",
  "name": "my-oak-api",
  "projectType": "application",
  "root": ".",
  "sourceRoot": "src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nx/deno:emit",
      "options": Object {
        "denoConfig": "deno.json",
        "main": "src/main.ts",
        "outputFile": "dist/my-oak-api/main.js",
      },
      "outputs": Array [
        "dist/my-oak-api",
      ],
    },
    "lint": Object {
      "executor": "@nx/deno:lint",
      "options": Object {
        "denoConfig": "deno.json",
      },
    },
    "serve": Object {
      "executor": "@nx/deno:run",
      "options": Object {
        "buildTarget": "my-oak-api:build",
      },
    },
    "test": Object {
      "executor": "@nx/deno:test",
      "options": Object {
        "coverageDirectory": "coverage/my-oak-api",
        "denoConfig": "deno.json",
      },
      "outputs": Array [
        "coverage/my-oak-api",
      ],
    },
  },
}
`;

exports[`Deno App Generator standalone should make an oak api with --framework=oak 2`] = `
"import { Application, Router } from 'https://deno.land/x/oak@v11.1.0/mod.ts';

const router = new Router();

router.get('/', (ctx) => {
  ctx.response.redirect('/api');
});

router.get('/api', (ctx) => {
  ctx.response.body = { message: 'Hello my-oak-api' };
  ctx.response.type = 'text/json';
});

const app = new Application();

app.addEventListener('listen', ({ hostname, port, secure }) => {
  console.log(
    \`Listening on: \${secure ? 'https://' : 'http://'}\${
      hostname ?? 'localhost'
    }:\${port}\`
  );
});

app.use(router.routes());
app.use(router.allowedMethods());

await app
  .listen({ port: Number(Deno.env.get('PORT') || 4200) })
  .catch((err) => {
    console.error('Error serving app. Original Error:', err);
  });
"
`;
